pulser-core/pulser/backend/state.py

```diff
@@ -242,3 +242,53 @@ def _to_abstract_repr(self) -> dict[str, Any]:
             "eigenstates": tuple(self._eigenstates),
             "amplitudes": dict(self._amplitudes),
         }
+
+
+class StateRepr(State):
+    """State subclass that supports serialization for remote backends."""
+
+    @classmethod
+    def _from_state_amplitudes(
+        cls,
+        *,
+        eigenstates: Sequence[Eigenstate],
+        amplitudes: Mapping[str, complex],
+    ) -> tuple[StateRepr, Mapping[str, complex]]:
+        """Implements the conversion used in `from_state_amplitudes()`."""
+        state = cls(eigenstates=eigenstates)
+        state._validate_amplitudes(
+            eigenstates=eigenstates, amplitudes=amplitudes
+        )
+        return state, amplitudes
+
+    def _to_abstract_repr(self) -> dict[str, Any]:
+        cls_name = self.__class__.__name__
+        if self._amplitudes is None:
+            raise AbstractReprError(
+                f"Failed to serialize state of type {cls_name!r} because it "
+                f"was not created via '{cls_name}.from_state_amplitudes()'."
+            )
+        return {
+            "eigenstates": tuple(self._eigenstates),
+            "amplitudes": dict(self._amplitudes),
+        }
+
+    @property
+    def n_qudits(self) -> int:
+        """``n_qudits`` not implemented in ``StateRepr``."""
+        raise NotImplementedError
```

Comment:
You can actually define this one through the amplitudes
```suggestion
        return self._validate_amplitudes(eigenstates=self._eigenstates, amplitudes=self._amplitudes)
```
Maybe you can just store this under `state._n_qudits` so you don't compute it every time

Reply:
Done by adding  a _n_qudits attribute to StateRepr